packagekit: Move CSS to less

martinpitt opened the pull request:
This does some meaningful grouping for easier understanding and
maintaining the styles.

martinpitt commented:
I was looking at less for #8752, and used this as an opportunity to learn a bit about it. There are no effective visual changes, and I visually verified the available updates list and automatic updates. The effective diff between the original and the less-built updates.css:

```diff
❱❱❱ diff -wBu /tmp/dist.pr/packagekit/updates.css dist/packagekit/updates.css
--- /tmp/dist.pr/packagekit/updates.css	2018-03-04 21:11:48.000000000 +0100
+++ dist/packagekit/updates.css	2018-03-04 21:49:52.000000000 +0100
@@ -1,4 +1,4 @@
-.tooltip-ct-outer {
+@import url(/page.css);.tooltip-ct-outer {
     display:  inline-block;
     position: relative;
 }
@@ -668,11 +666,15 @@
     max-width: 18ex;
     word-wrap: break-word;
 }
-
 tr.listing-ct-item td.type {
     white-space: nowrap;
 }
-
+tr.listing-ct-item td.changelog {
+  max-width: 60vw;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+}
 .severity-critical {
     color: #a30000;
 }
@@ -680,18 +681,9 @@
 .severity-important {
     color: #ec7a08;
 }
-
 .severity-low {
     color: #72767b;
 }
-
-tr.listing-ct-item td.changelog {
-    max-width: 60vw;
-    white-space: nowrap;
-    overflow: hidden;
-    text-overflow: ellipsis;
-}
-
 p.changelog {
     max-height: 20em;
     overflow: auto;
@@ -700,9 +691,9 @@
 p.changelog div.changelog {
     white-space: pre-wrap;
 }
-
 tr.security.listing-ct-item {
-    background-color: #fbf0f0 !important;  /* keep red background when expanded */
+  background-color: #fbf0f0 !important;
+  /* keep red background when expanded */
     border-top: 1px solid #e0e0e0;
     border-bottom: 1px solid #e0e0e0;
 }
@@ -782,16 +763,13 @@
     padding: 1ex;
     flex: none;
 }
-
 /* http://www.patternfly.org/pattern-library/communication/empty-state/ has a
  * gray background which we don't want here */
-
 .blank-slate-pf {
     background: inherit;
     border: none;
-}
-
 /* standard blank-slate-pf does not limit width of paragraph, which looks ugly */
+}
 .blank-slate-pf p {
     max-width: 50rem;
     margin: 0 auto;
@@ -824,20 +799,12 @@
     text-overflow: ellipsis;
     font-weight: 600;
 }
-
 .update-log {
     padding-top: 8em;
     margin: 0 auto;
     max-width: 69rem;
     text-align: center;
 }
-
-.update-log-content {
-    margin: 0 10ex;
-    height: 13em;
-    overflow-y: auto;
-}
-
 .update-log th {
     text-align: left;
     padding-right: 3ex;
@@ -842,11 +809,14 @@
     text-align: left;
     padding-right: 3ex;
 }
-
 .update-log td {
     text-align: left;
 }
-
+.update-log-content {
+  margin: 0 10ex;
+  height: 13em;
+  overflow-y: auto;
+}
 /* Layout fix for tooltip arrows.
  *
  * Sometimes when there are non-integer coordinates involved
@@ -860,12 +829,13 @@
 .tooltip.top .tooltip-arrow {
     bottom: 1px;
 }
-
 .tooltip.bottom .tooltip-arrow {
     top: 1px;
 }
-
-
+table.header-buttons {
+  margin-bottom: 20px;
+  /* replacement for h2's margin-bottom */
+}
 table.header-buttons td {
     vertical-align: middle;
 }
@@ -869,24 +839,17 @@
 table.header-buttons td {
     vertical-align: middle;
 }
-
 table.header-buttons h2 {
-    margin: 0 4rem 0 0; /* vertical default ones break vertical alignment */
-}
-
-table.header-buttons {
-    margin-bottom: 20px; /* replacement for h2's margin-bottom */
+  margin: 0 4rem 0 0;
+  /* vertical default ones break vertical alignment */
 }
-
 table.auto-conf {
     margin-bottom: 2em;
+  /* bootstrap sets widths to 100%, way too wide for Select widgets */
 }
-
 table.auto-conf td {
     padding-right: 1ex !important;
 }
-
-/* bootstrap sets widths to 100%, way too wide for Select widgets */
 table.auto-conf .dropdown-toggle {
     width: auto;
 }


```

mvollmer commented:
The `@import "/page.css"` in `pkg/packagekit/updates.css` now gets copied into the output, which leads to the browser trying to load `https://f27:9090/page.css` (e.g.), which doesn't exist.

From looking at other less sources, such as `shell/shell.less`, I think this should be changed to `@import "../lib/page.css"`, but I really have no idea how this all works.

martinpitt commented:
@mvollmer: Well spotted, thanks! Fixed. page.css gets included verbatim now, so the effective diff is bigger. I updated the diff in above comment.

mvollmer commented:
Now page.css gets included twice into `dist/packagekit/updates.css`.  Once by less, and I think  once more by webpack via autoupdates -> cockpit-component-select -> page.css.

I wouldn't know what to do about this in general.

mvollmer commented on pkg/packagekit/updates.less:
```diff
@@ -1,21 +1,27 @@
-@import "/page.css";
+@import (less) "../lib/page.css";
```
I now think this should be removed completely.  Sorry for the wrong advice earlier.

Nothing in updates.less needs anything from page.css, and by having it here causes page.css to be included twice.

martinpitt commented:
Indeed cockpit-components-select pulls in page.css [here](https://github.com/cockpit-project/cockpit/blob/master/pkg/lib/cockpit-components-select.jsx#L26).

martinpitt commented:
I dropped the import, updated the diff above (which now looks sane again), and visually verified that the page still looks correct. Thanks @mvollmer for pointing out this issue!
